sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1193,3 +1211,17 @@ case class Deduplicate(
 
   override def output: Seq[Attribute] = child.output
 }
+
+/** A trait used for logical plan nodes that create or replace V2 table definitions. */
+sealed trait V2TableCreation extends LogicalPlan {
```

Comment:
This name seems awkward to me. What about `CreateTablePlan` or `V2CreateTablePlan`?

Reply:
Also, why a sealed trait?

Reply:
doesn't need to be sealed. All the definitions were here anyway.